smithy-openapi/src/test/resources/software/amazon/smithy/openapi/fromsmithy/examples-test.openapi.json

```diff
@@ -0,0 +1,520 @@
+                  "9"
+                ]
+              }
+            }
+          },
+          {
+            "name": "barStringInputHeader",
+            "in": "header",
+            "schema": {
+              "type": "string"
+            },
+            "examples": {
+              "Bar_PATCH_example1": {
+                "summary": "Bar valid example",
+                "description": "barTestDoc",
+                "value": "barStringHeader1"
+              },
+              "Bar_PATCH_example2": {
+                "summary": "Bar error example",
+                "description": "barTestDoc2",
+                "value": "barStringHeader2"
+              }
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Bar 200 response",
+            "headers": {
+              "fooBarListOutputHeader": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  }
+                },
+                "examples": {
+                  "Bar_PATCH_example1": {
+                    "summary": "Bar valid example",
+                    "description": "barTestDoc",
+                    "value": [
+                      "barHeader1",
+                      "barHeader2",
+                      "barHeader3"
+                    ]
+                  }
+                }
+              },
+              "fooBarStringOutputHeader": {
+                "schema": {
+                  "type": "string"
+                },
+                "examples": {
+                  "Bar_PATCH_example1": {
+                    "summary": "Bar valid example",
+                    "description": "barTestDoc",
+                    "value": "barValidHeader"
+                  }
+                }
+              }
+            },
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/BarResponseContent"
+                },
+                "examples": {
+                  "Bar_PATCH_example1": {
+                    "summary": "Bar valid example",
+                    "description": "barTestDoc",
+                    "value": {
+                      "barOutputBody1": "fooBarValidBody3",
+                      "barOutputBody2": "fooBarValidBody4"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "FooBarError 400 response",
+            "headers": {
+              "Z-Foo": {
+                "schema": {
+                  "type": "string"
+                },
+                "examples": {
+                  "Bar_PATCH_example2": {
+                    "summary": "Bar error example",
+                    "description": "barTestDoc2",
+                    "value": "fooBarErrorHeader2"
+                  },
+                  "Foo_PUT_example2": {
```

Comment:
This seems wrong. Getting `Foo_PUT_example2` in the `Bar` operation's error.

Reply:
fixed by refactoring methods in `AbstractRestProtocol.java`. Needed the original `operation` used, so had to add that in as argument to every relevant method to make it usable by `createErrorExamples` and `createErrorBodyExamples`.